src/storage/storage.ts

```diff
@@ -344,6 +356,29 @@ export class Storage implements IStorage {
     await processHistory(this, { rewardLock: this.version?.reward_spend_min_blocks });
   }
 
+  async syncHistory(
+    startIndex: number,
+    count: number,
+    connection: FullNodeConnection,
+    shouldProcessHistory: boolean = false
+  ): Promise<void> {
+    if (
+      [HistorySyncMode.STREAM_XPUB, HistorySyncMode.STREAM_MANUAL].includes(this.historySyncMode)
+    ) {
+      const walletType = await this.getWalletType();
+      if (walletType !== WalletType.P2PKH) {
```

Comment:
What would be the other wallet types? I'm wondering why the stream API would not work for them.

Reply:
The other wallet type is P2SH for MultiSig wallets.
We could have support for them on manual stream mode but I believe we can add this support later.
What do you think?

Reply:
Got it. But wouldn't a better verification logic already include it? Something like:

```typescript
const supportedSyncMode = getSupportedSyncMode(this.getWalletType())
if not supportedSyncMode.includes(this.historySyncMode) {
    throw new Error('...')
}
```

Where:

```typescript
getSupportedSyncMode(walletType): set<historySyncMode> {
  if (walletType !== WalletType.P2PKH) {
    return set([HistorySyncMode.XPUB_STREAM_WS, HistorySyncMode.MANUAL_STREAM_WS])
  else if (walletType == WalletType.P2SH) {
    return set([HistorySyncMode.MANUAL_STREAM_WS])
  } else {
    throw new Error('invalid wallet type')
  }
}
```

Reply:
Done

Reply:
Thanks!